src/test/java/uk/gov/hmcts/reform/sscs/service/HearingsServiceRetryTest.java

```diff
@@ -0,0 +1,212 @@
+                .build();
+
+        wrapper = HearingWrapper.builder()
+            .state(CREATE_HEARING)
+            .caseData(caseData)
+            .caseData(caseData)
+            .build();
+
+        caseDetails = SscsCaseDetails.builder()
+            .data(caseData)
+            .build();
+    }
+
+    @DisplayName("When wrapper with a valid HearingResponse is given updateHearingResponse should return updated valid HearingResponse")
+    @ParameterizedTest
+    @CsvSource(value = {
+        "CREATE_HEARING,CREATE_HEARING",
+        "UPDATED_CASE,UPDATED_CASE",
+    }, nullValues = {"null"})
+    void updateHearingResponse(HearingState state, HearingEvent event) throws UpdateCaseException {
+        given(ccdCaseService.updateCaseData(
+            any(SscsCaseData.class),
+            any(EventType.class),
+            anyString(), anyString()))
+            .willReturn(caseDetails);
+
+        wrapper.setState(state);
+
+        HmcUpdateResponse response = HmcUpdateResponse.builder()
+            .versionNumber(VERSION)
+            .hearingRequestId(HEARING_REQUEST_ID)
+            .build();
+        assertThatNoException()
+            .isThrownBy(() -> hearingsService.hearingResponseUpdate(wrapper, response));
+
+        verify(ccdCaseService, times(1))
+            .updateCaseData(
+                any(SscsCaseData.class),
+                eq(event.getEventType()),
+                eq(event.getSummary()),
+                eq(event.getDescription()));
+    }
+
+
+    @DisplayName("When ccdCaseService throws UpdateCaseException, hearingResponseUpdate should retry "
+        + " and if then succeeds, throw no error not send additional messages to hmc")
+    @Test
+    void updateHearingResponse() throws UpdateCaseException {
+        given(ccdCaseService.updateCaseData(
+            any(SscsCaseData.class),
+            any(EventType.class),
+            anyString(),
+            anyString()))
+            .willThrow(UpdateCaseException.class)
+            .willReturn(caseDetails);
+
+        HmcUpdateResponse hmcUpdateResponse = HmcUpdateResponse.builder()
+            .hearingRequestId(HEARING_REQUEST_ID)
+            .versionNumber(1L)
+            .build();
+
+        given(hmcHearingApiService.sendCancelHearingRequest(
+            any(HearingCancelRequestPayload.class),
+            anyString()))
+            .willReturn(hmcUpdateResponse);
+
+        HmcUpdateResponse response = HmcUpdateResponse.builder()
+            .versionNumber(VERSION)
+            .hearingRequestId(HEARING_REQUEST_ID)
+            .build();
+
+        assertThatNoException()
+            .isThrownBy(() -> hearingsService.hearingResponseUpdate(wrapper, response));
+
+        verify(ccdCaseService, times(2))
+            .updateCaseData(
+                any(SscsCaseData.class),
+                any(EventType.class),
+                anyString(),
+                anyString());
+
+        verifyNoMoreInteractions(hmcHearingApiService);
```

Comment:
(Minor) could make this more specific

Reply:
Waqas advised to use verifyNoMoreInteractions where possible as it is more genetic/cleaner than using verify, but verifyNoMoreInteractions doesn't allow for specifying methods, unfortunately.
I don't mind either way.